src/library/basetracktablemodel.cpp

```diff
@@ -108,8 +108,7 @@ constexpr bool BaseTrackTableModel::kKeyColorsEnabledDefault;
 int BaseTrackTableModel::s_bpmColumnPrecision =
         kBpmColumnPrecisionDefault;
 bool BaseTrackTableModel::s_keyColorsEnabled = kKeyColorsEnabledDefault;
-ColorPalette BaseTrackTableModel::s_keyColorPalette =
-        mixxx::PredefinedColorPalettes::kDefaultKeyColorPalette;
+std::unique_ptr<ColorPalette> BaseTrackTableModel::s_pKeyColorPalette;
```

Comment:
any reason you chose a pointer over the more value-oriented `std::optional`?

Reply:
I just did not thought about it. A printer type is probably right, because in a static context  the memory allocated in a value type object is leaked. (Only esoteric reasons when using valgrind, because all memory is freed anyway on termination)

Reply:
I disagree... Why would the memory be leaked? The memory of the inner object has automatic storage duration, no cleanup needed. It can never be leaked if its not never dynamically allocated in the first place.

Reply:
Oh yes you are right. But `std::optional` is for me only a workaround for types without a default constructor. 
I would rather add this instead.

Reply:
Just realized that this will not work, because the colors are always considered as valid. I will use an `std::optional`.